src/domain/wallets/withdrawal-fee-calculator.ts

```diff
@@ -0,0 +1,10 @@
+import { toSats } from "@domain/bitcoin"
+
+export const WithdrawalFeeCalculator = (wallet: Wallet): WithdrawalFeeCalculator => {
```

Comment:
To match the `DepositFeeCalculator` setup I would probably make this so it can be called like:
```ts
WithdrawalFeeCalculator(onChainFee).onChainWithdrawalFee(wallet.withdrawFee)
```

The pattern here could be that the calculator constructor gets the btc/ln network amount to be operated on and the method takes any wallet-specific params required.

Reply:
uhmm we have two cases here, intraledger fee and onchain fee so the param in the constructor cant be onChainFee (the common part should be in the "constructor")

Reply:
I'm thinking of how to make this consistent with the `DepositFeeCalculator`

That one currently has a unique lifetime for each transaction and its common thing is the `amount` from the ln txn. The methods do operations on the amount using fee-related variables.

For the original comment, I was thinking the `onChainFee` might be the unique txn lifetime value and then methods take fee-related things like the wallet's withdrawFee constant to apply to the `onChainFee` in the constructor.

Having the common thing as the `wallet` could be tricky because we can probably get fee-related values from places like configs as well later on.

But I'm ok with changing this if we think it makes sense to broaden its lifetime. Maybe we can alternatively pass txn-related values to the methods as well and leave the constructor empty?
```ts
WithdrawalFeeCalculator().onChainWithdrawalFee({onChainFee, walletFee: wallet.withdrawFee})
```

Reply:
Fixed, will update DepositFeeCalculator in a different PR